### PR TITLE
Add tf inception v3

### PR DESCRIPTION
**Pull request by radcheb**
##### Issue
Implements local Inception v3 embedded

This contribution add a new local embedded for Inception v3 using TensorFlow. We use the same model used in Server embedded. The model we are using is available [here](http://download.tensorflow.org/models/image/imagenet/inception-2015-12-05.tgz).

##### Includes
- [X] Code changes
- [X] Tests
- [ ] Documentation

### CODE REVIEW

**Comment by PrimozGodec** [comment]
@radcheb thank you for your contribution. We were already considering to introduce local Inception before and then we did not decide for this step because we did not want to introduce TensorFlow as a dependency. Orange is a toll which must work on various platform and users are not computer science experts. Introducing TensorFlow causes issues on different (especially Windows) platforms, which is causing that Orange does not work out of the box anymore. 

This is also the reason why we made the custom NumPy implementation of SqueezeNet (which would work much faster with TensorFlow). 

Btw. Right now we are testing the new computation cluster for embedding which will significantly increase the speed of embedding. It will be available in next version of orange3-imageanalytics.

**Comment by radcheb** [comment]
> @radcheb thank you for your contribution. We were already considering to introduce local Inception before and then we did not decide for this step because we did not want to introduce TensorFlow as a dependency. Orange is a toll which must work on various platform and users are not computer science experts. Introducing TensorFlow causes issues on different (especially Windows) platforms, which is causing that Orange does not work out of the box anymore.
> 
> This is also the reason why we made the custom NumPy implementation of SqueezeNet (which would work much faster with TensorFlow).
> 
> Btw. Right now we are testing the new computation cluster for embedding which will significantly increase the speed of embedding. It will be available in next version of orange3-imageanalytics.

Thanks @PrimozGodec for your answer and sorry for the late response. Indeed Tensorflow was not well supported on all platforms. However, since Tensorflow 2.0, Windows support has been significantly improved. Thus I tried this plugin on Ubuntu, macOS, and Windows and it works perfectly with no further installations.

Actually, I'm already using this Tensorflow version in one of my projects and I just want to share it with the community. We developed it for may reasons. First, we wanted to have faster inference mainly when we run it on thousands of images. Second, we are also planning to introduce more models than inception v3 by taking advantage of Tensorflow Hub ready-to-use models: https://www.tensorflow.org/hub.
Finally, we want to go further and test a use case where we want to fine-tune our model with several labeled images and then use its embedding. Thus we need to have a light training layer which comes for free with Tensorflow Hub:
https://tfhub.dev/google/imagenet/inception_v3/feature_vector/4

I didn't know of the new cluster and I will be glad to contribute. Moreover, did you opensource the code for your actual cluster? We would like to see how the actual embedding is computed to be able to validate our implementation.

**Comment by PrimozGodec** [comment]
Thanks, @radcheb for the response. We are still a bit sceptic about Tensorflow. It is great that now it works on most of the devices, but if it does not work just for one type of CPU/OS our users will have issues (and they are not developers). On the other side, Tensorflow is a huge dependency (around 300MB) and I am not completely sure that we want that it comes together with the addon. 

But still, I was thinking about how to make it available locally. What do you think if we make Tensorflow as an optional dependency and if the user installs it in the environment the local inception shows up? I think it will improve the speed for users who have a GPU. For users without GPU embedding with our cluster is faster.

> I didn't know of the new cluster and I will be glad to contribute. Moreover, did you opensource the code for your actual cluster? We would like to see how the actual embedding is computed to be able to validate our implementation.

Code for our cluster is not opensource yet, since we have some other things on the same repository. I will think about making it opensource. For embedding we take the result of the layer of the network before the last layer: we take this tensor as an output: "InceptionV3/Logits/Dropout_1b/Identity:0".

**Comment by radcheb** [comment]
> Thanks, @radcheb for the response. We are still a bit sceptic about Tensorflow. It is great that now it works on most of the devices, but if it does not work just for one type of CPU/OS our users will have issues (and they are not developers). On the other side, Tensorflow is a huge dependency (around 300MB) and I am not completely sure that we want that it comes together with the addon.
> 
> But still, I was thinking about how to make it available locally. What do you think if we make Tensorflow as an optional dependency and if the user installs it in the environment the local inception shows up? I think it will improve the speed for users who have a GPU. For users without GPU embedding with our cluster is faster.
> 
> > I didn't know of the new cluster and I will be glad to contribute. Moreover, did you opensource the code for your actual cluster? We would like to see how the actual embedding is computed to be able to validate our implementation.
> 
> Code for our cluster is not opensource yet, since we have some other things on the same repository. I will think about making it opensource. For embedding we take the result of the layer of the network before the last layer: we take this tensor as an output: "InceptionV3/Logits/Dropout_1b/Identity:0".

Making local model available if tensorflow is installed is a great idea. I will work on it this weekend and test it locally before pinging you again. For GPU, I don't think we need it now since CPU inference is already fast. GPU will be more relevant for training.

**Comment by PrimozGodec** [comment]
@radcheb thank you for changes. It looks good. Can you update that PR such that Travis tests pass?

**Comment by PrimozGodec** [comment]
closing due to inactivity
